python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py

```diff
@@ -73,9 +74,20 @@ def __init__(
             client=client,
         )
 
-    async def generate_embeddings(self, texts: list[str], **kwargs: Any) -> ndarray:
+    async def generate_embeddings(
+        self,
+        texts: list[str],
+        settings: "PromptExecutionSettings | None" = None,
+        **kwargs: Any,
+    ) -> ndarray:
         """Generate embeddings from the Azure AI Inference service."""
-        settings: AzureAIInferenceEmbeddingPromptExecutionSettings = kwargs.get("settings", None)
+        if not settings:
+            settings = AzureAIInferenceEmbeddingPromptExecutionSettings()
+        else:
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, AzureAIInferenceEmbeddingPromptExecutionSettings)  # nosec
+        assert isinstance(self.client, EmbeddingsClient)  # nosec
```

Comment:
Would it be possible to set client in this class to the right type?

Reply:
The client is a union type but Mypy cannot narrow it down.